edgecontext.thrift

```diff
@@ -104,6 +114,24 @@ struct RequestId {
     1: string readable_id
 }
 
+/** Locale data from a request to our services that we want to
+propagate between services.
+
+This model is a component of the "Edge-Request" header.  You should not need to
+interact with this model directly, but rather through the EdgeRequestContext
+interface provided by baseplate.
+
+*/
+struct Locale {
+    /** The preferred locale code of the requesting client.
+    */
+    1: LanguageCode language_code
```

Comment:
please mark all fields as optional

Reply:
I've updated the Locale struct itself to be optional, since marking these as optional means we're dealing with pointers. Let me know if these should also be optional.

Reply:
🔕 Internally we have this guideline that all thrift struct fields should be marked as optional, but I just realized that we didn't mark some of the fields optional in `edgecontext.thrift`, so this is fine, at least it's consistent with existing `edgecontext.thrift`.

@spladug do you think we should change all fields inside this thrift to optional?

Reply:
I had to refresh myself on the rules for optional/required/"default" https://thrift.apache.org/docs/idl

The main difference between `optional` and the implicit default appears to be whether or not we serialize anything when the field is its default value / unset. It does appear that we serialize a rather large amount of nothing when we have a completely empty edge context right now so we could get some bytes-on-the-wire savings if we change things to `optional` (I think that's safe for Python, would that be breaking for Go because it changes the fields to pointers?) and at least we should stick to `optional` for new stuff going forward.

In this case, I think we'll probably want to make sure the struct is `optional`ed in the root struct, but the individual fields here could remain un-optional if we think they'll always be set together and that gives us better in-language ergonomics.

Reply:
So after combining the fields we have a single locale field that should always be present in some form, if the overall Locale struct has been provided (which is optional). Happy to change the field optional if that's the direction we're heading or leave as-is to line up with what we have here currently.